src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -303,7 +303,107 @@ class EngineSegment(BaseSegment):
         "ENGINE",
         Ref("EqualsSegment"),
         Sequence(
-            Ref("EngineFunctionSegment"),
+            Ref("TableEngineFunctionSegment"),
+            AnySetOf(
+                Sequence(
+                    "ORDER",
+                    "BY",
+                    OneOf(
+                        Ref("BracketedColumnReferenceListGrammar"),
+                        Ref("ColumnReferenceSegment"),
+                    ),
+                    optional=True,
+                ),
+                Sequence(
+                    "PARTITION",
+                    "BY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "PRIMARY",
+                    "KEY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "SAMPLE",
+                    "BY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "SETTINGS",
+                    Delimited(
+                        AnyNumberOf(
```

Comment:
This AnyNumberOf isn't needed, as Delimited covers that behaviour

Reply:
Fixed in commit `f56765e`.